### ISSUE TITLE
Phase 2: Semantic Affix Composition Training

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#60]
## Parent Issue
Part of #58 (Redesigned Training Plan)

## Goal
Train semantic affix embeddings (8 prefixes + 15 suffixes = 23 x 128d = 3K params) to learn how affixes transform root meanings.

## Semantic Affixes (from config/affix_classification.json)

**Prefixes (8)**: mal, re, ge, eks, pra, dis, for, mis
**Suffixes (15)**: ul, ej, il, aj, ar, et, eg, in, id, ec, ac, er, estr, um, obl

## Training Approach: Triplet Loss

```python
# anchor = "hundejo" embedding (composed)
# positive = "lernejo", "vendejo" (other -ej words)
# negative = "hundo" (base root without suffix)

loss = max(0, margin + sim(anchor, negative) - sim(anchor, positive))
```

## Implementation Tasks

- [ ] Freeze root embeddings from Phase 1
- [ ] Extract affix application patterns from corpus
- [ ] Implement TripletLoss training
- [ ] Evaluate affix cluster separation

## Success Criteria

1. All -ej words cluster together (places)
2. All -ul words cluster together (people)
3. embed("hund") + embed("-ej") approximately equals embed("hundejo")

## Dependencies
- Phase 1 must be complete (root embeddings frozen)
